src/components/Header/Header.jsx

```diff
@@ -139,105 +144,130 @@ const SettingsGroup = ({ children }) => {
-        rel="noopener noreferrer"
-      >
-        <LogOutButton />
-        <span>
-          {strings.app_logout}
-        </span>
-      </BugLink>
-    );
-
-    const { Announce } = this.state;
-
-    return (
-      <div>
-        <ToolbarHeader>
-          <VerticalAlignDiv>
-            <LogoGroup small={small} />
-            {small && <LinkGroup navbarPages={navbarPages} />}
-          </VerticalAlignDiv>
-          {!disableSearch && <SearchGroup />}
-          <VerticalAlignDiv style={{ marginLeft: '16px' }}>
-            {small && <AccountGroup />}
-            <SettingsGroup>
-              <LocalizationMenu />
-              <ReportBug />
-              {user ? <LogOut /> : null}
-            </SettingsGroup>
-          </VerticalAlignDiv>
-        </ToolbarHeader>
-        { location.pathname !== '/' && Announce && <Announce /> }
-      </div>
-    );
-  }
-}
+const LogoGroup = ({ onMenuClick }) => (
+  <div style={{ marginRight: 16 }}>
+    <VerticalAlignToolbar>
+      <MenuButtonWrapper>
+        <IconButton edge="start" color="inherit" onClick={onMenuClick}>
+          <MenuIcon />
+        </IconButton>
+      </MenuButtonWrapper>
+      <AppLogoWrapper>
+        <AppLogo />
+      </AppLogoWrapper>
+    </VerticalAlignToolbar>
+  </div>
+);
+
+LogoGroup.propTypes = {
+  onMenuClick: PropTypes.func,
+};
+
+const SearchGroup = () => (
+  <VerticalAlignToolbar style={{ marginLeft: 'auto' }}>
+    <ActionSearch style={{ marginRight: 6, opacity: '.6' }} />
+    <SearchForm />
+  </VerticalAlignToolbar>
+);
+
+const AccountGroup = () => (
+  <VerticalAlignToolbar>
+    <AccountWidget />
+  </VerticalAlignToolbar>
+);
+
+const ReportBug = ({ strings }) => (
+  <DropdownMenuItem component="a" href={REPORT_BUG_PATH} target="_blank" rel="noopener noreferrer" >
+    <BugReport style={{ marginRight: 32, width: 24, height: 24 }} />
+    {strings.app_report_bug}
+  </DropdownMenuItem>
+);
+
+ReportBug.propTypes = {
+  strings: PropTypes.shape({}),
+};
+
+const LogOut = ({ strings }) => (
+  <BugLink
+    href={`${process.env.REACT_APP_API_HOST}/logout`}
+    rel="noopener noreferrer"
+  >
+    <LogOutButton />
+    <span>
+      {strings.app_logout}
+    </span>
+  </BugLink>
+);
 
-const mapStateToProps = state => ({
-  small: state.browser.greaterThan.small,
-  user: state.app.metadata.data.user,
-  strings: state.app.strings,
-});
+LogOut.propTypes = {
+  strings: PropTypes.shape({}),
+};
+
+const Header = ({
+  location, disableSearch,
+}) => {
+  const [Announce, setAnnounce] = useState(null);
+  const [menuIsOpen, setMenuState] = useState(false);
+  const navigationState = useContext(navigationContext);
+  const small = useSelector(state => state.browser.greaterThan.small);
+  const user = useSelector(state => state.app.metadata.data.user);
+  const strings = useSelector(state => state.app.strings);
+
+  useEffect(() => {
+    import('../Announce').then(ann => setAnnounce(ann.default));
+  }, []);
```

Comment:
This is special. `useEffect` can be used to handle stuff like initialization, value updates etc.

This effect basically is a `componentDidMount` because it will only run once on the first render call and afterwards won't called again. Thats because we passed an empty dependency array to it (see line 222).

Inside the effect we set the announceState to the imported Announce component.

This would look like this in a class:

```js
componentDidMount() {
  import('../Announce').then(ann => this.setState({
    Announce: ann.default,
  }))
}
```